rofs/cache.go

```diff
@@ -8,10 +8,16 @@ import (
 	"syscall"
 
 	"github.com/threefoldtech/0-fs/meta"
+	"github.com/threefoldtech/0-fs/storage"
 )
 
-func (fs *filesystem) path(hash string) string {
-	base := fs.cache
+type Cache struct {
```

Comment:
Add a NewCache(path, storage) function to create a new instance of cache. This way you don't need to expose the internals of it